sqlmesh/core/plan/builder.py

```diff
@@ -429,15 +442,35 @@ def _categorize_snapshots(
                         # previous version for non-seed models.
                         # New snapshots of seed models are considered non-breaking ones.
                         if not snapshot.is_seed:
+                            if (
```

Comment:
This check seems to be repeated over and over again. Consider moving it inside `_validate_schema_change`